frankenphp.c

```diff
@@ -302,7 +302,7 @@ uintptr_t frankenphp_request_shutdown()
 			int i;
 
 			for (i=0; i<NUM_TRACK_VARS; i++) {
-				zval_ptr_dtor(&PG(http_globals)[i]);
+				zval_ptr_dtor_nogc(&PG(http_globals)[i]);
 			}
 		} zend_end_try();
 	}
```

Comment:
Re-adding a call to `php_hash_environment()` after this line prevents the crash (but is likely not a good fix).